lib/Models/MapInteractionMode.js

```diff
@@ -24,6 +24,12 @@ function MapInteractionMode(options) {
     this.onCancel = options.onCancel;
 
     /**
+     * Opportunity to display extra custom UI elements.
+     * @type {Function}
+     */
+    this.customUi = defaultValue(options.customUi, function() { return ""; });
```

Comment:
This is pretty interesting.  From the standpoint of the `MapInteractionMode`, this is kind of like an opaque object.  In our React-based UI, it's a function that returns a React component.  In our old Knockout based UI, it might be a view-model or something like that, so not a function at all.  It might be good to document it this way.  Something like:

```
/**
 * Gets or sets the details of a custom user interface for this map interaction mode.  This property is not
 * used by the `MapInteractionMode` itself, so it can be anything that is suitable for the user interface.  In
 * the standard React-based user interface included with TerriaJS, this property is a function that is called
 * with no parameters and is expected to return a React component.
 * @type {Any}
 */
```

Also the default value should probably be undefined.